### PR TITLE
Update prometheus dependency to commit #8836

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: yeya24 <yb532204897@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This pr updates Prometheus dependency to https://github.com/prometheus/prometheus/commit/0a8912433a457b54a871df75e72afc3e45a31d5c.

Split from pr https://github.com/thanos-io/thanos/pull/4239.

This update included commit https://github.com/prometheus/prometheus/pull/8723, which breaks the existing TSDB store test. The writability and readability are not deterministic so I removed most of the tests. Let me know if there is a better way to solve this.

### CODE REVIEW

**Comment by yeya24** [comment]
This testcase https://github.com/thanos-io/thanos/blob/main/pkg/store/bucket_test.go#L1938 is broken due to the recent 120 chunks split change in Prometheus.

Any idea about how to fix or change it? @metalmatze @bwplotka @onprem

**Comment by bwplotka** [comment]
Hah, well looks like we relied on this ;p

> 
// This method relies on a bug in TSDB Compactor which will just merge overlapping chunks into one big chunk.// If compactor is fixed in the future, we may need a different way of generating the block, or commit// existing block to the repository.

I think, maybe it's time to remove all tests relying on such large chunk situation, since. we hopefully don't have such?

File: pkg/store/tsdb_test.go [hunk 1]
```diff
@@ -684,49 +649,6 @@ func TestTSDBStore_SeriesAccessWithoutDelegateClosing(t *testing.T) {
 					_ = string(c.Raw.Data) // Access bytes by converting them to different type.
 				}))
 			}
-			testutil.NotOk(t, testutil.FaultOrPanicToErr(func() {
```
Comment (bwplotka) [inline pkg/store/tsdb_test.go:687]: Why those are removed?

File: pkg/store/tsdb_test.go [hunk 2]
```diff
@@ -595,26 +580,6 @@ func TestTSDBStore_SeriesAccessWithDelegateClosing(t *testing.T) {
 		// Let's close pending querier!
 		testutil.Equals(t, 1, len(csrv.closers))
 		testutil.Ok(t, csrv.closers[0].Close())
-
-		// Expect flush and close to be unblocked and without errors.
```
Comment (bwplotka) [inline pkg/store/tsdb_test.go:599]: Why those are removed?

**Review by bwplotka — COMMENTED** [review]
LGTM, just I am worried about other removed tests around Flush.. why?

File: pkg/store/tsdb_test.go (hunk 2 above)
Comment (bwplotka) [reply pkg/store/tsdb_test.go:599]: hmmm prometheus/prometheus#8723 ... Can we skip them instead 🤔

Comment (yeya24) [reply pkg/store/tsdb_test.go:599]: Yes, that's a better option.

**Comment by yeya24** [comment]
@bwplotka PR updated, let me know if I need to update anything else.

**Review by kakkoyun — APPROVED** [review]
LGTM 🎉 🚀

**Comment by yeya24** [comment]
Issue opened #4266. Will merge it.

**Pull request merged**